system/serial.h

```diff
@@ -0,0 +1,181 @@
+#ifndef _SERIAL_REG_H
+#define _SERIAL_REG_H
+/**
+ * \file
+ *
+ * Provides the TTY interface. It provides an 80x25 VT100 compatible
+ * display via Serial/UART.
+ *
+ *//*
+ * Copyright (C) 2004-2022 Sam Demeulemeester.
+ */
+
+
+#define SERIAL_DEFAULT_BAUDRATE 115200
+#define SERIAL_DEFAULT_BITS     8
+#define SERIAL_DEFAULT_PARITY   0
+
+#define SERIAL_PORT_0x3F8   0
+#define SERIAL_PORT_0x2F8   1
+#define SERIAL_PORT_0x3E8   2
+#define SERIAL_PORT_0x2E8   3
+
+static const uint16_t serial_base_ports[] = { 0x3f8, 0x2f8, 0x3e8, 0x2e8 };
+
+struct serial_port {
+	bool enable;
+	bool is_mmio;
+	unsigned int parity;
```

Comment:
Could use narrower data types for these, but it wouldn't have much of an impact on the code size.